src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/Tensor.cs

```diff
@@ -691,6 +691,56 @@ private Tensor<T> GetTriangle(int offset, bool upper)
         /// <param name="value">The new value to set at the specified position in this Tensor.</param>
         public abstract void SetValue(int index, T value);
 
+        /// <summary>
+        /// The type that implements enumerators for <see cref="Tensor{T}"/> instances.
+        /// </summary>
+        public struct Enumerator : IEnumerator<T>
+        {
+            private readonly Tensor<T> tensor;
+            private int index;
+
+            internal Enumerator(Tensor<T> tensor)
+            {
+                Debug.Assert(tensor != null);
+
+                this.tensor = tensor;
+                index = 0;
+                Current = default;
+            }
+
+            public T Current { get; private set; }
+
+            public bool MoveNext()
+            {
+                if (index < tensor.Length)
+                {
+                    Current = tensor.GetValue(index);
+                    ++index;
+                    return true;
+                }
+                else
+                {
+                    Current = default;
+                    return false;
+                }
+            }
+
+            object? IEnumerator.Current => Current;
+
+            void IEnumerator.Reset()
+            {
+                index = 0;
+                Current = default;
```

Comment:
Could be
```suggestion
                this = default;
```
too.

Reply:
I did not know you could do that. I'll update it.